milli/src/update/settings.rs

```diff
@@ -6,12 +6,51 @@ use chrono::Utc;
 use grenad::CompressionType;
 use itertools::Itertools;
 use rayon::ThreadPool;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
 
+use crate::{FieldsIdsMap, Index};
 use crate::criterion::Criterion;
 use crate::facet::FacetType;
-use crate::update::index_documents::{Transform, IndexDocumentsMethod};
 use crate::update::{ClearDocuments, IndexDocuments, UpdateIndexingStep};
-use crate::{Index, FieldsIdsMap};
+use crate::update::index_documents::{IndexDocumentsMethod, Transform};
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Setting<T> {
+    Set(T),
+    Reset,
+    NotSet,
+}
+
+impl<T> Default for Setting<T> {
+    fn default() -> Self {
+        Self::NotSet
+    }
+}
+
+impl<T> Setting<T> {
+    pub const fn is_not_set(&self) -> bool {
+        matches!(*self, Self::NotSet)
```

Comment:
```suggestion
        matches!(self, Self::NotSet)
```

I think this dereferencing is not mandatory.